snuba/query/expressions.py

```diff
@@ -110,10 +110,88 @@ def visit_lambda(self, exp: Lambda) -> TVisited:
         raise NotImplementedError
 
 
+class StringifyVisitor(ExpressionVisitor[str]):
+    """Visitor implementation to turn an expression into a string format
+
+    Usage:
+        # Any expression class supported by the visitor will do
+        >>> exp: Expression = Expression()
+        >>> visitor = StringifyVisitor()
+        >>> exp_str = exp.accept(visitor)
+    """
+
+    def __init__(self) -> None:
+        # keeps track of the level of the AST we are currently in, this is necessary for nice indentation
+        # before recursively going into subnodes increment this counter, decrement it after the recursion
+        # is done
```

Comment:
Nit, while we are not too strict on line length, I'd suggest to keep docstrings and comment shorter as per PEP8 so they are more readable when reviewing.
I generally stay within 70-75 characters for comments, but we are not strict about it.